### PR TITLE
Pull Request Commit Comments

### PR DESCRIPTION
**Pull request by jonnyarnold**
Samuel will check comments on PR commit comments, as well as PR comments.

Thank you Github for making both of these comment lists of a similar format.

### CODE REVIEW

**Comment by reevoo-samuel** [comment]
*If you're new, read the [guidelines](https://github.com/reevoo/guidelines/blob/master/pull_requests.md).*

Good code is **tested**, **designed for change**, **easy to understand** and **able to cope when
things go wrong**.

Good pull requests are **descriptive**, **small** and **short-lived**.

**For the Author**
- [ ] Description explains what you have done and why.
- [ ] Tests have been written (or you have explained why you don't need
tests).
- [ ] Documentation/comments have been added/updated.

**For the Reviewer**
- [ ] The code is well-designed and able to cope with change.
- [ ] The code is easy to understand.
- [ ] Tests cover happy and unhappy paths in the code.
- [ ] Tests are green on CI (or pulled/tested on your machine).

File: lib/samuel/checks/has_comments.ex [hunk 1]
```diff
@@ -33,16 +33,19 @@ defmodule Samuel.Checks.HasComments do
...
+    data.commit_comments ++ data.pr_comments
+  end
 
   # Returns the number of comments made by someone who is not the Pull Request
   # author or Samuel.
   defp other_user_comments(data) do
     users_that_dont_count = users_that_dont_count(data.event)
 
-    data.comments
+    all_comments(data)
```
Comment (lpil) [inline lib/samuel/checks/has_comments.ex]:
```elixir
data
|> all comments
|> etc
```

File: lib/samuel/checks/has_comments.ex [hunk 2]
```diff
@@ -33,16 +33,19 @@ defmodule Samuel.Checks.HasComments do
...
   end
 
+  defp all_comments(data) do
+    data.commit_comments ++ data.pr_comments
+  end
 
   # Returns the number of comments made by someone who is not the Pull Request
   # author or Samuel.
   defp other_user_comments(data) do
     users_that_dont_count = users_that_dont_count(data.event)
```
Comment (lpil) [inline lib/samuel/checks/has_comments.ex]: Do we want variables shadowing function names?

File: lib/samuel/checks/has_comments.ex (hunk 1 above)
Comment (jonnyarnold) [inline lib/samuel/checks/has_comments.ex]: Good spot

File: lib/samuel/checks/has_comments.ex (hunk 2 above)
Comment (jonnyarnold) [inline lib/samuel/checks/has_comments.ex]: I don't think this is particularly harmful - the shadowing occurs in a small context (the function).

Comment (lpil) [inline lib/samuel/checks/has_comments.ex]: Is there any benefit?

Comment (jonnyarnold) [inline lib/samuel/checks/has_comments.ex]: Yes - they represent the same data, and now they have similar names.

Comment (lpil) [inline lib/samuel/checks/has_comments.ex]: I challenge that, one's data, one's data transformation, but you call.

Comment (jonnyarnold) [inline lib/samuel/checks/has_comments.ex]: It's one of my favourite things about ruby: if you see `x = foo` you don't know if `foo` is a method or a variable, and *the caller doesn't care* - it just wants the data.

Comment (lpil) [inline lib/samuel/checks/has_comments.ex]: In Elixir functions and variable live in two different namespaces, we don't have objects with mutable state you can use for memoization, and it's not Haskell with lazy evaluation, so we want to be explicit about where and when we do computation.

Comment (errm) [inline lib/samuel/checks/has_comments.ex]: I somewhat agree with @lphil, this kind of thing ultimately leads to confusion.

**Comment by errm** [comment]
broadly looks good to me bwtfdik

**Pull request merged**